1051 - add initial git commit signing instructions and example images

sfmcgee opened the pull request:
<!-- readthedocs-preview civicactions-handbook start -->
----
:books: Documentation preview :books:: https://civicactions-handbook--1101.org.readthedocs.build/en/1101/

<!-- readthedocs-preview civicactions-handbook end -->

sfmcgee committed:
1051 - add initial git commit signing instructions and example images

openprivacy commented on practice-areas/engineering/git.md:
```diff
@@ -147,6 +147,151 @@ Here is a recommended Git configuration block that it is recommended to add to y
+
+Commit signing allows for identification and non-repudiation of individual commits to repositories in both Github and Gitlab. It improves security of our repositories by ensuring we know who made changes. The following instructions describe how to create a new GPG keypair, associate it with your Github and Gitlab accounts, and enable commit signing by default with git.
+
+### Creating a new GPG Key
```
This section is excellent and IMO should get its own page ("Setting up GnuPG") for GPG in general, and here keep just the lines that are specific to the `.gitconfig` setup and other Git-related commands. I would link to the general GnuPG page from https://guidebook.civicactions.com/en/latest/common-practices-tools/security/encryption/#the-gnu-privacy-guard-gpg

sfmcgee replied:
Ok, thanks @openprivacy! I split it out into a new gnupg.md page. Let me know if it works.

openprivacy commented on practice-areas/engineering/git.md:
```diff
@@ -147,6 +147,151 @@ Here is a recommended Git configuration block that it is recommended to add to y
+gpg: depth: 0  valid:   1  signed:   0  trust: 0-, 0q, 0n, 0m, 0f, 1u
+/home/username/.gnupg/pubring.kbx
+-------------------------------
+sec   rsa4096/FF94245366AB39B2 2022-09-13 [SC]
```
Even though it is just a fingerprint, consider randomizing the values here (and maybe blur it out in the assets above). You might want to blur our the "mcgee" part of your name, too -- this is a public page. Up to you (unless there is Guidebook guidance on this.)

ContentBrewmaster replied:
@sfmcgee did you want to consider any of @openprivacy's comments before we merge this?

sfmcgee replied:
@ContentBrewmaster Yes, actually I would, thank you. I just haven't had a chance to work on those updates yet.

sfmcgee replied:
@openprivacy - thanks for the suggestion, I updated the fingerprints and the images.

ContentBrewmaster committed:
Update git.md

Completed copy edit:
- sentence case for headings
- replaced hyphen punctuation for new sentences with periods and capital letters.
- replaced e.g. with "such as"
- capitalized Git throughout, except in code blocks
- made all headings imperative

sfmcgee committed:
1051 - address openprivacys peer review and split out gpg setup

dmundra commented on common-practices-tools/security/gnupg.md:
@sfmcgee you want to add this file to the navigation in the 'mkdocs.yml' file.

sfmcgee replied:
Thanks @dmundra , updated.

sfmcgee committed:
1051 - add gnupg.md to mkdocs.yml and fix a typo

dmundra committed:
Merge branch 'master' into 1051-add-instructions-to-configure-git-commit-signing-to-the-guidebook-smcgee

openprivacy reviewed (approved):
@sfmcgee This looks great! Thank you for making these changes.
